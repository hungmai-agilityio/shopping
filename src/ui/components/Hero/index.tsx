import { ReactNode } from 'react';
import { clsx } from 'clsx';

// Components
import Breadcrumb, { BreadcrumbItem } from '@/ui/components/common/BreadCrumb';

interface HeroProps {
  isHeroSecond?: boolean;
  isBreadCrumb?: boolean;
  styles?: string;
  children: ReactNode;
  items?: BreadcrumbItem[];
}

const Hero = ({
  isHeroSecond = false,
  isBreadCrumb,
  styles,
  children,
  items
}: HeroProps) => {
  return (
    <div
      className={clsx(
        'relative w-full h-full overflow-hidden py-10 px-28',
        styles
      )}
    >
      {isBreadCrumb && (
        <Breadcrumb
          items={items!}
          color="text-white"
          styles="relative container z-20 md:flex hidden"
        />
      )}
      <div className="relative z-10">{children}</div>

      <div
        className={clsx('absolute inset-0 bg-gradient-primary ', {
          'top-0 left-0 right-0 bottom-0 -skew-y-6 origin-top-left max-h-hero':
            isHeroSecond
        })}
      ></div>
    </div>
  );
};

export default Hero;
