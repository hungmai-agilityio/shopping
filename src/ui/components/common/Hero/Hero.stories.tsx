import { mockBreadCrumb } from '@/mocks';
import Hero from '@/ui/components/common/Hero';
import { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Hero> = {
  component: Hero,
  tags: ['autodocs'],
  argTypes: {
    children: {
      description: 'Content to render inside Hero'
    },

    isBreadCrumb: {
      description: 'Show isBreadCrumb with data "items'
    },

    isHeroSecond: {
      description: 'Render another UI of Hero',
      control: { type: 'boolean' }
    },
    items: {
      description: 'Breadcrumb data, including name and path',
      control: { type: 'object' }
    },

    styles: {
      description: 'Custom css'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Hero>;

export const Default: Story = {
  args: {
    items: mockBreadCrumb,
    children: <p className="text-white text-xl font-bold">Children</p>,
    isBreadCrumb: true
  }
};

export const Second: Story = {
  args: {
    items: mockBreadCrumb,
    children: <p className="text-white text-xl font-bold">Children</p>,
    isBreadCrumb: true,
    isHeroSecond: true
  }
};
