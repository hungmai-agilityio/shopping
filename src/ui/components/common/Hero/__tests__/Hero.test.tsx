import { mockBreadCrumb } from '@/mocks';
import Hero from '@/ui/components/common/Hero';
import { render, screen } from '@testing-library/react';

describe('Hero component', () => {
  it('render Hero with default props', () => {
    render(<Hero children="Test" />);

    expect(screen.getByText('Test')).toHaveClass('relative z-10');
  });

  it('render Hero with BreakCrumb', () => {
    render(<Hero children="Test" isBreadCrumb items={mockBreadCrumb} />);

    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Products')).toBeInTheDocument();
    expect(screen.getByText('Phones')).toBeInTheDocument();
  });

  it('should be render match to snapshot', () => {
    const { container } = render(<Hero children="Test" />);

    expect(container).toMatchSnapshot();
  });
});
