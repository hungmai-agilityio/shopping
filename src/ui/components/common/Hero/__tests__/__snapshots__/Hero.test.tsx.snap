// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component should be render match to snapshot 1`] = `
<div>
  <div
    class="relative w-full h-full overflow-hidden py-10 px-28"
  >
    <div
      class="relative z-10"
    >
      Test
    </div>
    <div
      class="absolute inset-0 bg-gradient-primary "
    />
  </div>
</div>
`;
